### ISSUE TITLE
code should use "res" or "rval" but not both

### ISSUE THREAD

**Issue by dankelley** [dankelley/oce#811]
I'm doing an awful lot of code twiddling for the `unit` branch, and so I'm looking at code written in dribs and drabs over years. Sometimes a return value is spelled `rval` and other times it is `res`. This seems nutty but from the below, a lot would be changed in making it uniform. @richardsc -- what do you think, should I leave this dual-name system or switch? Since the word is always followed by certain non-letter characters, it wouldn't really be that hard for me to change the whole thing to `res` or `rval`. Do you prefer one over the other? (My vote is to switch to `res` but I don't exactly know why, other than the history. If I could find some R coding resource that suggested a particular name, I'd be inclined to go that way...)
```
R[units*]$ git grep rval |wc
     936    4811   56967
R[units*]$ git grep res |wc
    2107   13860  161722
```

**Comment by richardsc** [comment]
I really don't think it's that big of an issue, considering that these are only internal names within functions and are not returned. I understand it's probably a pain when going through lots of different functions as you're doing for units now, but I'm tempted to just leave them. Most of the work now is done in single functions, so the difference wouldn't really be apparent anyway.

**Comment by dankelley** [comment]
You're right -- it's for the convenience of development. I just spent half an hour tracing a bug to that.  I think from now on, all new code will use `res` and whenever I'm altering code anyway, I'll switch to `res`. For example, there are many hundreds of lines of codes changed in `units` (in a trivial way -- not scary but just a bunch of simple pattern edits on dozen-line chunks) and since they are all changed, I don't see a problem in changing them again slightly before I commit. In my experience, looking at code usually is a good thing, and I don't do file-wide edits, but rather in dozen-line chunks so my eye can verify things.

**Comment by dankelley** [comment]
PS. `git grep` is wonderful on these things because it colourizes so you can see at a glance what edit pattern to use.
<img width="1026" alt="screen shot 2015-12-23 at 3 50 31 pm" src="https://cloud.githubusercontent.com/assets/99469/11983822/fe3ea1ea-a98c-11e5-808b-9ab7156d23fa.png">

**Comment by dankelley** [comment]
Not to be cantankerous, but I decided that since I wrote most of the code, my vote is a bit stronger. Also, I wanted the chance to skim through the text by hitting the "y" key repeatedly to agree to the changes. (As with emacs, vim has a nice command that changes patterns only if they are classifiable as a 'word', which is the case for these variables.)  I didn't full read that whole code, but it's quite good to get an overview, and to see patterns of coding changes. The code looks quite nice, actually. It's not beautiful, but it's certainly not junk, either.

I won't push for a few days since I'm not going in to the office for a while. I'll do commits of course, so the log will make sense when I do commit. I'm quite amazed at the size of the code change in this `units` branch, but in a way, not. During term and during proposal-writing times, I am too scared to do much more than timid bug fixes. (And, yes, tomorrow I'll pop over to `develop` and fix that colormap bug for you.)
